integration-tests/environments/react-native/src/App.js

```diff
@@ -24,132 +24,152 @@
-                    delete this.client;
-                });
-            },
+  prepareTests() {
+    this.client = new MochaRemoteClient({
+      id: Platform.OS,
+      onConnected: () => {
+        console.log('Connected to mocha-remote-server');
+        this.setState({status: 'waiting'});
+      },
+      onDisconnected: ({reason}) => {
+        console.error(`Disconnected: ${reason}`);
+        this.setState({status: 'disconnected', reason});
+      },
+      onInstrumented: mocha => {
+        // Setting the title of the root suite
+        global.title = `React-Native on ${Platform.OS}`;
+        // Provide the global Realm constructor to the tests
+        // Simply requiring Realm will set the global for us ...
+        // global.Realm = require('realm');
+        require('realm');
+        global.fs = require('react-native-fs');
+        global.path = require('path-browserify');
+        global.environment = {
+          reactNative: Platform.OS,
+          android: Platform.OS === 'android',
+          ios: Platform.OS === 'ios',
+        };
+        // Make all test related modules reinitialize
+        const modules = require.getModules();
+        for (const [id, m] of Object.entries(modules)) {
+          if (m.verboseName.indexOf('realm-integration-tests') !== -1) {
+            m.isInitialized = false;
+          }
+        }
+        // Require in the integration tests
+        require('realm-integration-tests');
+        beforeEach(() => {
+          // Adding an async task before each, allowing the UI to update
+          return new Promise(resolve => setTimeout(resolve, 0));
         });
-    }
+      },
+      onRunning: runner => {
+        this.setState({
+          status: 'running',
+          failures: 0,
+          currentTestIndex: 0,
+        });
+        runner.on('test', test => {
+          // Compute the current test index - incrementing it if we're running
+          // Set the state to update the UI
+          this.setState({
+            status: 'running',
+            currentTest: test.fullTitle(),
+            currentTestIndex: this.state.currentTestIndex + 1,
+            totalTests: runner.total,
+          });
+        });
+        runner.on('end', () => {
+          this.setState({
+            status: 'ended',
+            failures: runner.failures,
+          });
+          // Stop trying to connect to the remote server
+          this.client.disconnect();
+          delete this.client;
+        });
+      },
+    });
+  }
 }
 
 const styles = StyleSheet.create({
-    container: {
-        flex: 1,
-        justifyContent: "center",
-        alignItems: "center",
-        backgroundColor: "#F5FCFF",
-    },
-    status: {
-        fontSize: 20,
-        textAlign: "center",
-        margin: 10,
-    },
-    details: {
-        fontSize: 14,
-        textAlign: "center",
-        margin: 10,
-    },
+  container: {
+    flex: 1,
+    justifyContent: 'center',
```

Comment:
is this some tool that is causing this

Reply:
ESlint and [Prettier](https://prettier.io/) I suggest we make this more explicit and align the tools across, once we have multiple (sub) packages relying on it.

Reply:
The thing is we might not be able to align all tools to work exactly the same on all platforms and environments we have. That's why having automating on save is problematic since it will toggle the formatting depending on which platform the change is done. I think in practice this will not work and will be an inconvenience for us.